src/G2/Execution/Rules.hs

```diff
@@ -1106,22 +1110,55 @@ addExtConds s ng e1 ais e2 stck =
     in
     ([strue, sfalse], ng)
 
--- | Inject binds into the eenv. The LHS of the [(Id, Expr)] are treated as
--- seed values for the names.
-liftBinds :: [(Id, Expr)] -> E.ExprEnv -> Expr -> NameGen ->
+-- | This function aims to extract the types variable and types as determined by the coercion in the pattern 
+-- n1 represent the type variables 
+-- n2 represent the type determined by the coercion
+-- for example: in a GADT when we want to establish a coercion between n and zero
+-- In this case, n1 represent n and n2 represent zero
```

Comment:
This comment will appear in Hackage documentation, and thus should be readable independently of the code.  But currently, it relies on the code to be understandable.  Writing can also broadly be tightened up.

"This function aims to extract pairs of types being coerced between.   Given a coercion `t1 :~ t2`, the tuple (t1, t2) is returned."